pkg/segment/writer/packer.go

```diff
@@ -380,8 +380,12 @@ func (ss *SegStore) encodeSingleDictArray(arraykey string, data []byte, maxIdx u
 				finalErr = fmt.Errorf("encodeSingleDictArray : received unknown key  %+s", keyType)
 			}
 			if bi != nil {
-				bi.uniqueWordCount += addToBlockBloom(bi.Bf, []byte(keyName))
-				bi.uniqueWordCount += addToBlockBloom(bi.Bf, []byte(keyVal))
+				bytesKeyName := []byte(keyName)
+				bytesKeyVal := []byte(keyVal)
+				bi.uniqueWordCount += addToBlockBloom(bi.Bf, bytesKeyName)
+				bi.uniqueWordCount += addToBlockBloom(bi.Bf, bytesKeyVal)
+				bi.uniqueWordCount += addToBlockBloom(bi.Bf, utils.BytesToLowerInPlace(bytesKeyName))
+				bi.uniqueWordCount += addToBlockBloom(bi.Bf, utils.BytesToLowerInPlace(bytesKeyVal))
```

Comment:
The new changes that I have done on this PR.

Reply:
This looks good, but there's one more thing I just found. Here you're doing a copy to convert a string to `[]byte`:
```
bytesKeyVal := []byte(keyVal)
```
But `keyVal` is from a call to this function: https://github.com/siglens/siglens/blob/develop/pkg/segment/writer/packer.go#L403-L427 and that function gets a `[]byte` for the `keyVal` and copies it to a `string`. So I think you can make that function return a `[]byte` instead to avoid a copy. And same for `keyName`

But also be mindful you're using these a little lower: https://github.com/siglens/siglens/pull/1536/commits/53066ae4331dd47d3be9462f6dbf473dc77dab3a#diff-f71656920dd3349225bbb82c8a0daf499efcd2971346afb1c050faaa83a874d3R390 so you don't want to accidentally change the case where they're used there

Reply:
BTW, you don't have to point out the new changes for this PR; github let's you select a single commit or a range of commits to review, so I just selected your second commit since the initial commit was reverting something

Reply:
oh Yes, I forgot about that.

Reply:
> But also be mindful you're using these a little lower ... so you don't want to accidentally change the case where they're used there

Should I tag the function name as `UnSafe`? Refactor the function name to `UnsafeBytesToLowerInPlace` or `UnsafeBytesToLower`?

Reply:
No, I think what you have currently `BytesToLowerInPlace` is clear already